supervisor/iptablesctrl/update_portsets.go

```diff
@@ -0,0 +1,126 @@
+package iptablesctrl
+
+import (
+	"bufio"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/aporeto-inc/trireme/cache"
+	"github.com/bvandewalle/go-ipset/ipset"
+)
+
+const (
+	procNetTCPFile                    = "/proc/net/tcp"
+	portSetUpdateIntervalMilliseconds = 500
+	portEntryTimeout                  = 60
+	uidFieldOffset                    = 7
+	procHeaderLineNum                 = 1
+	portOffset                        = 1
+	ipPortOffset                      = 1
+	sockStateOffset                   = 3
+	sockListeningState                = "0A"
+	hexFormat                         = 16
+	integerSize                       = 64
+)
+
+func getUIDPortSetMappings(i *Instance, uid string) (interface{}, error) {
+	return i.UIDToPortSet.Get(uid)
+}
+
+func getUIDPortMappings(i *Instance, uid string) (interface{}, error) {
+	return i.UIDToPorts.Get(uid)
+}
```

Comment:
You should not return an `interface{}` type here, you should cast the return value here. (make sure to verify if you can cast to the type you want first. A bad cast results to a panic)